src/transformers/modeling_utils.py

```diff
@@ -2678,10 +2678,18 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     " make sure you have the latest version of `bitsandbytes` installed"
                 )
 
-            model = replace_with_bnb_linear(
+            model, has_been_replaced = replace_with_bnb_linear(
```

Comment:
Raise the warning directly in `replace_with_bnb_linear` instead of returning this extra flag.